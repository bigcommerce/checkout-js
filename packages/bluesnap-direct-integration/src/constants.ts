export enum BluesnapECPAccountType {
    ConsumerChecking = 'CONSUMER_CHECKING',
    ConsumerSavings = 'CONSUMER_SAVINGS',
    CorporateChecking = 'CORPORATE_CHECKING',
    CorporateSavings = 'CORPORATE_SAVINGS',
}
