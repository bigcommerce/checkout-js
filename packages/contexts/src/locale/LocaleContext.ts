import type { CurrencyService, LanguageService } from '@bigcommerce/checkout-sdk';
import { createContext } from 'react';

export interface LocaleContextType {
    language: LanguageService;
    date?: {
        inputFormat: string;
    };
    currency?: CurrencyService;
}

export const LocaleContext = createContext<LocaleContextType | undefined>(undefined);
