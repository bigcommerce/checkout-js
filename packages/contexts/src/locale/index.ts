export { LocaleContext, type LocaleContextType } from './LocaleContext';
export { default as LocaleProvider } from './LocaleProvider';
export { useLocale } from './useLocale';
