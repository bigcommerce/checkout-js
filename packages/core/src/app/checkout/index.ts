export { RenderCheckout, RenderCheckoutOptions } from './renderCheckout';
export { default as CheckoutContext, CheckoutContextProps } from './CheckoutContext';
export {
    default as CheckoutProvider,
    CheckoutProviderProps,
    CheckoutProviderState,
} from './CheckoutProvider';
export { default as withCheckout, WithCheckoutProps } from './withCheckout';
export { default as CheckoutSupport } from './CheckoutSupport';
export { default as NoopCheckoutSupport } from './NoopCheckoutSupport';
export { default as navigateToOrderConfirmation } from './navigateToOrderConfirmation';
