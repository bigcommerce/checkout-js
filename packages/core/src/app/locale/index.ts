export { default as LocaleContext, LocaleContextType } from './LocaleContext';
export { default as createLocaleContext } from './createLocaleContext';
export { default as getDefaultTranslations } from './getDefaultTranslations';
export { default as getLanguageService, InitializeLanguageService } from './getLanguageService';
export { default as masterpassFormatLocale } from './masterpassFormatLocale';
export { default as withCurrency, WithCurrencyProps } from './withCurrency';
export { default as withLanguage, WithLanguageProps } from './withLanguage';
export { default as withDate, WithDateProps } from './withDate';
export { default as LanguageWindow, isLanguageWindow } from './LanguageWindow';
export { default as LocaleProvider } from './LocaleProvider';
export { default as TranslatedHtml, TranslatedHtmlProps } from './TranslatedHtml';
export { default as TranslatedLink, TranslatedLinkProps } from './TranslatedLink';
export { default as TranslatedString, TranslatedStringProps } from './TranslatedString';
