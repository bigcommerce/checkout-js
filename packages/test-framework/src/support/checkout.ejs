<!DOCTYPE html>
<html>
    <head>
        <title>Local Testing Environment</title>
        <base href="/" target="_top"/>
        <meta name="viewport" content="width=device-width, initial-scale=1, maximum-scale=1" />
        <link href="https://fonts.googleapis.com/css?family=Karla:400|Montserrat:400,500,700&display=swap" rel="stylesheet" />
    </head>
    <body>
    <header class="checkoutHeader optimizedCheckout-header">
        <div class="checkoutHeader-content">
            <h1 class="is-srOnly">Checkout</h1>
            <h2 class="checkoutHeader-heading">
                <a class="checkoutHeader-link" href="<%= storeUrl %>/">
                    <span class="header-logo-text">Local Testing Environment</span>
                </a>
            </h2>
        </div>
    </header>
    <div class="ng-checkout-container" id="micro-app-ng-checkout">
        <div class="ng-checkout-view" data-ui-view></div>
    </div>
    <script type="text/javascript">
        window.checkoutConfig = {
            checkoutId: '<%= checkoutId %>',
            containerId: 'micro-app-ng-checkout',
        };
    </script>
    <script src="<%= localhostUrl %>/auto-loader.js" type="text/javascript"></script>
    <script type="text/javascript">
        (function() {
            window.checkout.renderCheckout({
                checkoutId: '<%= checkoutId %>',
                containerId: 'micro-app-ng-checkout',
                sentryConfig: null,
            })
        })();
    </script>
    </body>
</html>
