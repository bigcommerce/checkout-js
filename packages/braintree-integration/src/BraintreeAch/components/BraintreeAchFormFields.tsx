import { LanguageService } from '@bigcommerce/checkout-sdk';
import React, { FunctionComponent, useCallback } from 'react';

import { useLocale } from '@bigcommerce/checkout/locale';
import { usePaymentFormContext } from '@bigcommerce/checkout/payment-integration-api';
import { DynamicFormField } from '@bigcommerce/checkout/ui';

import {
    BraintreeAchFieldType,
    businessBraintreeAchFormFields,
    OwnershipTypes,
    personalBraintreeAchFormFields,
} from '../constants';

const isBraintreeAchFormFieldName = (fieldName: string): fieldName is BraintreeAchFieldType => {
    return Object.values(BraintreeAchFieldType).includes(fieldName as BraintreeAchFieldType);
};

const getTranslatedLabelByFieldName = (fieldName: string, language: LanguageService) => {
    const braintreeAchFormLabelsMap = {
        firstName: language.translate('address.first_name_label'),
        lastName: language.translate('address.last_name_label'),
        accountNumber: language.translate('payment.account_number_label'),
        routingNumber: language.translate('payment.account_routing_label'),
        businessName: language.translate('payment.business_name_label'),
        ownershipType: language.translate('payment.ownership_type_label'),
        accountType: language.translate('payment.account_type_label'),
    };

    return isBraintreeAchFormFieldName(fieldName)
        ? braintreeAchFormLabelsMap[fieldName]
        : fieldName;
};

const BraintreeAchFormFields: FunctionComponent = () => {
    const { language } = useLocale();
    const { paymentForm } = usePaymentFormContext();
    const { getFieldValue, setFieldValue } = paymentForm;

    const ownershipTypeValue = getFieldValue(BraintreeAchFieldType.OwnershipType);
    const fieldValues =
        ownershipTypeValue === OwnershipTypes.Business
            ? businessBraintreeAchFormFields
            : personalBraintreeAchFormFields;

    const handleChange = useCallback(
        (fieldId: string) => (value: string) => {
            setFieldValue(fieldId, value);
        },
        [setFieldValue],
    );

    return (
        <>
            {fieldValues.map((field) => (
                <DynamicFormField
                    extraClass={`dynamic-form-field--${field.id}`}
                    field={field}
                    key={field.id}
                    label={getTranslatedLabelByFieldName(field.name, language)}
                    onChange={handleChange(field.id)}
                />
            ))}
        </>
    );
};

export default BraintreeAchFormFields;
