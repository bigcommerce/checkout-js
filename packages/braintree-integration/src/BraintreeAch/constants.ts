import { FormField } from '@bigcommerce/checkout-sdk';

import { DynamicFormFieldType } from '@bigcommerce/checkout/ui';

export enum BraintreeAchFieldType {
    BusinessName = 'businessName',
    AccountType = 'accountType',
    AccountNumber = 'accountNumber',
    RoutingNumber = 'routingNumber',
    OwnershipType = 'ownershipType',
    FirstName = 'firstName',
    LastName = 'lastName',
}

export enum OwnershipTypes {
    Personal = 'Personal',
    Business = 'Business',
}

export enum AccountTypes {
    Savings = 'Savings',
    Checking = 'Checking',
}

const accountTypeOptions = [
    {
        value: AccountTypes.Savings,
        label: AccountTypes.Savings,
    },
    {
        value: AccountTypes.Checking,
        label: AccountTypes.Checking,
    },
];

export const ownershipTypeOptions = [
    {
        value: OwnershipTypes.Personal,
        label: OwnershipTypes.Personal,
    },
    {
        value: OwnershipTypes.Business,
        label: OwnershipTypes.Business,
    },
];

export const defaultBraintreeAchFormFields: FormField[] = [
    {
        name: BraintreeAchFieldType.AccountType,
        custom: false,
        id: BraintreeAchFieldType.AccountType,
        label: 'Account Type',
        required: true,
        fieldType: DynamicFormFieldType.DROPDOWM,
        options: {
            items: accountTypeOptions,
        },
    },
    {
        name: BraintreeAchFieldType.AccountNumber,
        custom: false,
        id: BraintreeAchFieldType.AccountNumber,
        label: 'Account Number',
        required: true,
        max: 9,
        min: 8,
    },
    {
        name: BraintreeAchFieldType.RoutingNumber,
        custom: false,
        id: BraintreeAchFieldType.RoutingNumber,
        label: 'Routing Number',
        required: true,
    },
    {
        name: BraintreeAchFieldType.OwnershipType,
        custom: false,
        id: BraintreeAchFieldType.OwnershipType,
        label: 'Ownership Type',
        required: true,
        fieldType: DynamicFormFieldType.DROPDOWM,
        options: {
            items: ownershipTypeOptions,
        },
    },
];

export const personalBraintreeAchFormFields: FormField[] = [
    ...defaultBraintreeAchFormFields,
    {
        name: BraintreeAchFieldType.FirstName,
        custom: false,
        id: BraintreeAchFieldType.FirstName,
        label: 'First Name',
        required: true,
    },
    {
        name: BraintreeAchFieldType.LastName,
        custom: false,
        id: BraintreeAchFieldType.LastName,
        label: 'Last Name',
        required: true,
    },
];

export const businessBraintreeAchFormFields: FormField[] = [
    ...defaultBraintreeAchFormFields,
    {
        name: BraintreeAchFieldType.BusinessName,
        custom: false,
        id: BraintreeAchFieldType.BusinessName,
        label: 'Business Name',
        required: true,
    },
];
