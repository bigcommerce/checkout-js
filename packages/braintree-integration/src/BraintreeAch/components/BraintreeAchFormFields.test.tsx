import { render } from '@testing-library/react';
import { Formik } from 'formik';
import { noop } from 'lodash';
import React from 'react';

import { createLocaleContext, LocaleContext } from '@bigcommerce/checkout/locale';
import {
    PaymentFormContext,
    PaymentFormService,
} from '@bigcommerce/checkout/payment-integration-api';
import { getPaymentFormServiceMock, getStoreConfig } from '@bigcommerce/checkout/test-utils';

import { OwnershipTypes } from '../constants';

import BraintreeAchFormFields from './BraintreeAchFormFields';

describe('BraintreeAchFormFields Component', () => {
    let paymentForm: PaymentFormService;

    beforeEach(() => {
        paymentForm = getPaymentFormServiceMock();
    });

    it('should render form with personal fields', () => {
        jest.spyOn(paymentForm, 'getFieldValue').mockReturnValue(OwnershipTypes.Personal);

        const view = render(
            <Formik initialValues={{}} onSubmit={noop}>
                <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                    <PaymentFormContext.Provider value={{ paymentForm }}>
                        <BraintreeAchFormFields />
                    </PaymentFormContext.Provider>
                </LocaleContext.Provider>
            </Formik>,
        );

        expect(view).toMatchSnapshot();
    });

    it('should render form with business fields', () => {
        jest.spyOn(paymentForm, 'getFieldValue').mockReturnValue(OwnershipTypes.Business);

        const view = render(
            <Formik initialValues={{}} onSubmit={noop}>
                <LocaleContext.Provider value={createLocaleContext(getStoreConfig())}>
                    <PaymentFormContext.Provider value={{ paymentForm }}>
                        <BraintreeAchFormFields />
                    </PaymentFormContext.Provider>
                </LocaleContext.Provider>
            </Formik>,
        );

        expect(view).toMatchSnapshot();
    });
});
