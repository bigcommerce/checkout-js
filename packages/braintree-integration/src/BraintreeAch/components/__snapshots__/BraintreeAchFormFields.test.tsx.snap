// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BraintreeAchFormFields Component should render form with business fields 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dynamic-form-field dynamic-form-field--accountType"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountType"
            id="accountType-label"
          >
            Account Type
          </label>
          <div
            class="dropdown-chevron"
          >
            <div
              class="icon"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <select
            aria-labelledby="accountType-label accountType-field-error-message"
            class="form-select optimizedCheckout-form-select"
            data-test="accountType-select"
            id="accountType"
            name="accountType"
          >
            <option
              value="Savings"
            >
              Savings
            </option>
            <option
              value="Checking"
            >
              Checking
            </option>
          </select>
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--accountNumber"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountNumber"
            id="accountNumber-label"
          >
            Account Number
          </label>
          <input
            aria-labelledby="accountNumber-label accountNumber-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="accountNumber-text"
            id="accountNumber"
            max="9"
            min="8"
            name="accountNumber"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--routingNumber"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="routingNumber"
            id="routingNumber-label"
          >
            Routing Number
          </label>
          <input
            aria-labelledby="routingNumber-label routingNumber-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="routingNumber-text"
            id="routingNumber"
            name="routingNumber"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--ownershipType"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="ownershipType"
            id="ownershipType-label"
          >
            Ownership Type
          </label>
          <div
            class="dropdown-chevron"
          >
            <div
              class="icon"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <select
            aria-labelledby="ownershipType-label ownershipType-field-error-message"
            class="form-select optimizedCheckout-form-select"
            data-test="ownershipType-select"
            id="ownershipType"
            name="ownershipType"
          >
            <option
              value="Personal"
            >
              Personal
            </option>
            <option
              value="Business"
            >
              Business
            </option>
          </select>
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--businessName"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="businessName"
            id="businessName-label"
          >
            Business Name
          </label>
          <input
            aria-labelledby="businessName-label businessName-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="businessName-text"
            id="businessName"
            name="businessName"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dynamic-form-field dynamic-form-field--accountType"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="accountType"
          id="accountType-label"
        >
          Account Type
        </label>
        <div
          class="dropdown-chevron"
        >
          <div
            class="icon"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <select
          aria-labelledby="accountType-label accountType-field-error-message"
          class="form-select optimizedCheckout-form-select"
          data-test="accountType-select"
          id="accountType"
          name="accountType"
        >
          <option
            value="Savings"
          >
            Savings
          </option>
          <option
            value="Checking"
          >
            Checking
          </option>
        </select>
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--accountNumber"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="accountNumber"
          id="accountNumber-label"
        >
          Account Number
        </label>
        <input
          aria-labelledby="accountNumber-label accountNumber-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="accountNumber-text"
          id="accountNumber"
          max="9"
          min="8"
          name="accountNumber"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--routingNumber"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="routingNumber"
          id="routingNumber-label"
        >
          Routing Number
        </label>
        <input
          aria-labelledby="routingNumber-label routingNumber-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="routingNumber-text"
          id="routingNumber"
          name="routingNumber"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--ownershipType"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="ownershipType"
          id="ownershipType-label"
        >
          Ownership Type
        </label>
        <div
          class="dropdown-chevron"
        >
          <div
            class="icon"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <select
          aria-labelledby="ownershipType-label ownershipType-field-error-message"
          class="form-select optimizedCheckout-form-select"
          data-test="ownershipType-select"
          id="ownershipType"
          name="ownershipType"
        >
          <option
            value="Personal"
          >
            Personal
          </option>
          <option
            value="Business"
          >
            Business
          </option>
        </select>
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--businessName"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="businessName"
          id="businessName-label"
        >
          Business Name
        </label>
        <input
          aria-labelledby="businessName-label businessName-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="businessName-text"
          id="businessName"
          name="businessName"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`BraintreeAchFormFields Component should render form with personal fields 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="dynamic-form-field dynamic-form-field--accountType"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountType"
            id="accountType-label"
          >
            Account Type
          </label>
          <div
            class="dropdown-chevron"
          >
            <div
              class="icon"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <select
            aria-labelledby="accountType-label accountType-field-error-message"
            class="form-select optimizedCheckout-form-select"
            data-test="accountType-select"
            id="accountType"
            name="accountType"
          >
            <option
              value="Savings"
            >
              Savings
            </option>
            <option
              value="Checking"
            >
              Checking
            </option>
          </select>
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--accountNumber"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="accountNumber"
            id="accountNumber-label"
          >
            Account Number
          </label>
          <input
            aria-labelledby="accountNumber-label accountNumber-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="accountNumber-text"
            id="accountNumber"
            max="9"
            min="8"
            name="accountNumber"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--routingNumber"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="routingNumber"
            id="routingNumber-label"
          >
            Routing Number
          </label>
          <input
            aria-labelledby="routingNumber-label routingNumber-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="routingNumber-text"
            id="routingNumber"
            name="routingNumber"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--ownershipType"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="ownershipType"
            id="ownershipType-label"
          >
            Ownership Type
          </label>
          <div
            class="dropdown-chevron"
          >
            <div
              class="icon"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <select
            aria-labelledby="ownershipType-label ownershipType-field-error-message"
            class="form-select optimizedCheckout-form-select"
            data-test="ownershipType-select"
            id="ownershipType"
            name="ownershipType"
          >
            <option
              value="Personal"
            >
              Personal
            </option>
            <option
              value="Business"
            >
              Business
            </option>
          </select>
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--firstName"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="firstName"
            id="firstName-label"
          >
            First Name
          </label>
          <input
            aria-labelledby="firstName-label firstName-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="firstName-text"
            id="firstName"
            name="firstName"
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        class="dynamic-form-field dynamic-form-field--lastName"
      >
        <div
          class="form-field"
        >
          <label
            class="form-label optimizedCheckout-form-label"
            for="lastName"
            id="lastName-label"
          >
            Last Name
          </label>
          <input
            aria-labelledby="lastName-label lastName-field-error-message"
            class="form-input optimizedCheckout-form-input"
            data-test="lastName-text"
            id="lastName"
            name="lastName"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="dynamic-form-field dynamic-form-field--accountType"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="accountType"
          id="accountType-label"
        >
          Account Type
        </label>
        <div
          class="dropdown-chevron"
        >
          <div
            class="icon"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <select
          aria-labelledby="accountType-label accountType-field-error-message"
          class="form-select optimizedCheckout-form-select"
          data-test="accountType-select"
          id="accountType"
          name="accountType"
        >
          <option
            value="Savings"
          >
            Savings
          </option>
          <option
            value="Checking"
          >
            Checking
          </option>
        </select>
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--accountNumber"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="accountNumber"
          id="accountNumber-label"
        >
          Account Number
        </label>
        <input
          aria-labelledby="accountNumber-label accountNumber-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="accountNumber-text"
          id="accountNumber"
          max="9"
          min="8"
          name="accountNumber"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--routingNumber"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="routingNumber"
          id="routingNumber-label"
        >
          Routing Number
        </label>
        <input
          aria-labelledby="routingNumber-label routingNumber-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="routingNumber-text"
          id="routingNumber"
          name="routingNumber"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--ownershipType"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="ownershipType"
          id="ownershipType-label"
        >
          Ownership Type
        </label>
        <div
          class="dropdown-chevron"
        >
          <div
            class="icon"
          >
            <svg
              height="24"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </div>
        </div>
        <select
          aria-labelledby="ownershipType-label ownershipType-field-error-message"
          class="form-select optimizedCheckout-form-select"
          data-test="ownershipType-select"
          id="ownershipType"
          name="ownershipType"
        >
          <option
            value="Personal"
          >
            Personal
          </option>
          <option
            value="Business"
          >
            Business
          </option>
        </select>
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--firstName"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="firstName"
          id="firstName-label"
        >
          First Name
        </label>
        <input
          aria-labelledby="firstName-label firstName-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="firstName-text"
          id="firstName"
          name="firstName"
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      class="dynamic-form-field dynamic-form-field--lastName"
    >
      <div
        class="form-field"
      >
        <label
          class="form-label optimizedCheckout-form-label"
          for="lastName"
          id="lastName-label"
        >
          Last Name
        </label>
        <input
          aria-labelledby="lastName-label lastName-field-error-message"
          class="form-input optimizedCheckout-form-input"
          data-test="lastName-text"
          id="lastName"
          name="lastName"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
